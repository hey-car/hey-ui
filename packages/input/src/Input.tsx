import React, { FocusEvent, useCallback, useState } from 'react';
import cn from 'classnames';

import FormControl from '@heycar-uikit/form-control';

import { InputProps } from './Input.types';

import styles from './styles/default.module.css';

const Input = React.forwardRef<HTMLInputElement, InputProps>(
  (
    {
      value,
      type = 'text',
      hint,
      pattern,
      error,
      label,
      fullWidth = false,
      disabled,
      readOnly,
      rightIcon,
      rightAddons,
      leftIcon,
      onFocus,
      onBlur,
      onChange,
      onInput,
      className,
      dataTestId,
      ...restProps
    },
    ref,
  ) => {
    const inputClassNames = cn(styles.input, className, {
      [styles.hasLeftIcon]: leftIcon,
    });
    const ariaLabel = typeof label === 'string' ? label : undefined;
    const [isFocused, setFocused] = useState(restProps.autoFocus);

    const isFilled = Boolean(value ? value : false);
    const handleInputFocus = useCallback(
      (event: FocusEvent<HTMLInputElement>) => {
        if (!readOnly) setFocused(true);
        if (onFocus) onFocus(event);
      },
      [onFocus, readOnly],
    );

    return (
      <FormControl
        disabled={disabled}
        error={error}
        filled={isFilled || isFocused}
        focused={isFocused}
        fullWidth={fullWidth}
        hint={hint}
        label={label}
        leftIcon={leftIcon}
        rightAddons={rightAddons}
        rightIcon={rightIcon}
      >
        <input
          {...restProps}
          aria-label={ariaLabel}
          className={inputClassNames}
          data-test-id={dataTestId}
          disabled={disabled}
          onBlur={onBlur}
          onChange={onChange}
          onFocus={handleInputFocus}
          onInput={onInput}
          pattern={pattern}
          readOnly={readOnly}
          ref={ref}
          type={type}
          value={value}
        />
      </FormControl>
    );
  },
);

Input.displayName = 'Input';

export default Input;
