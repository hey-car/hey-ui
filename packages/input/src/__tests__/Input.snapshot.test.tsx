import React from 'react';
import { render } from '@testing-library/react';

import Input from '../Input';

describe('Input Snapshots tests', () => {
  it('should match callback methods', () => {
    expect(
      render(
        <Input
          onChange={jest.fn()}
          onClick={jest.fn()}
          onMouseDown={jest.fn()}
          onMouseUp={jest.fn()}
          value="value"
        />,
      ),
    ).toMatchSnapshot();
  });

  it('should match label and value props', () => {
    expect(render(<Input label="Label" value="value" />)).toMatchSnapshot();
  });

  it('should match left and right icons', () => {
    expect(
      render(
        <Input label="Label" leftIcon="ðŸš—" rightIcon="ðŸš—" value="value" />,
      ),
    ).toMatchSnapshot();
  });

  it('should match disabled state', () => {
    expect(
      render(<Input disabled={true} label="Label" value="value" />),
    ).toMatchSnapshot();
  });

  it('should match error hint/state', () => {
    expect(
      render(
        <Input
          error="This is the placeholder for error note"
          label="Label"
          value="value"
        />,
      ),
    ).toMatchSnapshot();
  });

  it('should match error hint', () => {
    expect(
      render(
        <Input
          hint="This is the placeholder for error note"
          label="Label"
          value="value"
        />,
      ),
    ).toMatchSnapshot();
  });

  it('should match fullWidth', () => {
    expect(
      render(<Input fullWidth={true} label="Label" value="value" />),
    ).toMatchSnapshot();
  });
});
