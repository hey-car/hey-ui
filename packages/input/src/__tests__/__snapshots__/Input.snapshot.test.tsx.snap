// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Input Snapshots tests should match callback methods 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div
        class="component"
      >
        <div
          class="inner filled"
        >
          <div
            class="inputWrapper"
          >
            <div
              class="input"
            >
              <input
                class="input"
                type="text"
                value="value"
              />
            </div>
          </div>
          <fieldset
            aria-hidden="true"
            class="fieldset"
          >
            <legend
              class="legend"
            />
          </fieldset>
        </div>
      </div>
    </div>
  </body>,
  "container": <div>
    <div
      class="component"
    >
      <div
        class="inner filled"
      >
        <div
          class="inputWrapper"
        >
          <div
            class="input"
          >
            <input
              class="input"
              type="text"
              value="value"
            />
          </div>
        </div>
        <fieldset
          aria-hidden="true"
          class="fieldset"
        >
          <legend
            class="legend"
          />
        </fieldset>
      </div>
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;

exports[`Input Snapshots tests should match disabled state 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div
        class="component disabled"
      >
        <div
          class="inner filled"
        >
          <div
            class="inputWrapper"
          >
            <label
              class="label"
            >
              <span
                class="labelInner"
              >
                Label
              </span>
            </label>
            <div
              class="input"
            >
              <input
                aria-label="Label"
                class="input"
                disabled=""
                type="text"
                value="value"
              />
            </div>
          </div>
          <fieldset
            aria-hidden="true"
            class="fieldset"
          >
            <legend
              class="legend"
            >
              <span>
                Label
              </span>
            </legend>
          </fieldset>
        </div>
      </div>
    </div>
  </body>,
  "container": <div>
    <div
      class="component disabled"
    >
      <div
        class="inner filled"
      >
        <div
          class="inputWrapper"
        >
          <label
            class="label"
          >
            <span
              class="labelInner"
            >
              Label
            </span>
          </label>
          <div
            class="input"
          >
            <input
              aria-label="Label"
              class="input"
              disabled=""
              type="text"
              value="value"
            />
          </div>
        </div>
        <fieldset
          aria-hidden="true"
          class="fieldset"
        >
          <legend
            class="legend"
          >
            <span>
              Label
            </span>
          </legend>
        </fieldset>
      </div>
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;

exports[`Input Snapshots tests should match error hint 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div
        class="component"
      >
        <div
          class="inner filled"
        >
          <div
            class="inputWrapper"
          >
            <label
              class="label"
            >
              <span
                class="labelInner"
              >
                Label
              </span>
            </label>
            <div
              class="input"
            >
              <input
                aria-label="Label"
                class="input"
                type="text"
                value="value"
              />
            </div>
          </div>
          <fieldset
            aria-hidden="true"
            class="fieldset"
          >
            <legend
              class="legend"
            >
              <span>
                Label
              </span>
            </legend>
          </fieldset>
        </div>
        <span
          class="sub hint"
        >
          This is the placeholder for error note
        </span>
      </div>
    </div>
  </body>,
  "container": <div>
    <div
      class="component"
    >
      <div
        class="inner filled"
      >
        <div
          class="inputWrapper"
        >
          <label
            class="label"
          >
            <span
              class="labelInner"
            >
              Label
            </span>
          </label>
          <div
            class="input"
          >
            <input
              aria-label="Label"
              class="input"
              type="text"
              value="value"
            />
          </div>
        </div>
        <fieldset
          aria-hidden="true"
          class="fieldset"
        >
          <legend
            class="legend"
          >
            <span>
              Label
            </span>
          </legend>
        </fieldset>
      </div>
      <span
        class="sub hint"
      >
        This is the placeholder for error note
      </span>
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;

exports[`Input Snapshots tests should match error hint/state 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div
        class="component hasError"
      >
        <div
          class="inner filled"
        >
          <div
            class="inputWrapper"
          >
            <label
              class="label"
            >
              <span
                class="labelInner"
              >
                Label
              </span>
            </label>
            <div
              class="input"
            >
              <input
                aria-label="Label"
                class="input"
                type="text"
                value="value"
              />
            </div>
          </div>
          <fieldset
            aria-hidden="true"
            class="fieldset"
          >
            <legend
              class="legend"
            >
              <span>
                Label
              </span>
            </legend>
          </fieldset>
        </div>
        <span
          class="sub error"
          role="alert"
        >
          This is the placeholder for error note
        </span>
      </div>
    </div>
  </body>,
  "container": <div>
    <div
      class="component hasError"
    >
      <div
        class="inner filled"
      >
        <div
          class="inputWrapper"
        >
          <label
            class="label"
          >
            <span
              class="labelInner"
            >
              Label
            </span>
          </label>
          <div
            class="input"
          >
            <input
              aria-label="Label"
              class="input"
              type="text"
              value="value"
            />
          </div>
        </div>
        <fieldset
          aria-hidden="true"
          class="fieldset"
        >
          <legend
            class="legend"
          >
            <span>
              Label
            </span>
          </legend>
        </fieldset>
      </div>
      <span
        class="sub error"
        role="alert"
      >
        This is the placeholder for error note
      </span>
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;

exports[`Input Snapshots tests should match fullWidth 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div
        class="component fullWidth"
      >
        <div
          class="inner filled"
        >
          <div
            class="inputWrapper"
          >
            <label
              class="label"
            >
              <span
                class="labelInner"
              >
                Label
              </span>
            </label>
            <div
              class="input"
            >
              <input
                aria-label="Label"
                class="input"
                type="text"
                value="value"
              />
            </div>
          </div>
          <fieldset
            aria-hidden="true"
            class="fieldset"
          >
            <legend
              class="legend"
            >
              <span>
                Label
              </span>
            </legend>
          </fieldset>
        </div>
      </div>
    </div>
  </body>,
  "container": <div>
    <div
      class="component fullWidth"
    >
      <div
        class="inner filled"
      >
        <div
          class="inputWrapper"
        >
          <label
            class="label"
          >
            <span
              class="labelInner"
            >
              Label
            </span>
          </label>
          <div
            class="input"
          >
            <input
              aria-label="Label"
              class="input"
              type="text"
              value="value"
            />
          </div>
        </div>
        <fieldset
          aria-hidden="true"
          class="fieldset"
        >
          <legend
            class="legend"
          >
            <span>
              Label
            </span>
          </legend>
        </fieldset>
      </div>
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;

exports[`Input Snapshots tests should match label and value props 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div
        class="component"
      >
        <div
          class="inner filled"
        >
          <div
            class="inputWrapper"
          >
            <label
              class="label"
            >
              <span
                class="labelInner"
              >
                Label
              </span>
            </label>
            <div
              class="input"
            >
              <input
                aria-label="Label"
                class="input"
                type="text"
                value="value"
              />
            </div>
          </div>
          <fieldset
            aria-hidden="true"
            class="fieldset"
          >
            <legend
              class="legend"
            >
              <span>
                Label
              </span>
            </legend>
          </fieldset>
        </div>
      </div>
    </div>
  </body>,
  "container": <div>
    <div
      class="component"
    >
      <div
        class="inner filled"
      >
        <div
          class="inputWrapper"
        >
          <label
            class="label"
          >
            <span
              class="labelInner"
            >
              Label
            </span>
          </label>
          <div
            class="input"
          >
            <input
              aria-label="Label"
              class="input"
              type="text"
              value="value"
            />
          </div>
        </div>
        <fieldset
          aria-hidden="true"
          class="fieldset"
        >
          <legend
            class="legend"
          >
            <span>
              Label
            </span>
          </legend>
        </fieldset>
      </div>
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;

exports[`Input Snapshots tests should match left and right icons 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div
        class="component hasLeftIcon"
      >
        <div
          class="inner filled"
        >
          <div
            class="addons leftIcon"
          >
            🚗
          </div>
          <div
            class="inputWrapper"
          >
            <label
              class="label"
            >
              <span
                class="labelInner"
              >
                Label
              </span>
            </label>
            <div
              class="input"
            >
              <input
                aria-label="Label"
                class="input hasLeftIcon"
                type="text"
                value="value"
              />
            </div>
          </div>
          <fieldset
            aria-hidden="true"
            class="fieldset"
          >
            <legend
              class="legend"
            >
              <span>
                Label
              </span>
            </legend>
          </fieldset>
          <div
            class="addons rightIcon"
          >
            🚗
          </div>
        </div>
      </div>
    </div>
  </body>,
  "container": <div>
    <div
      class="component hasLeftIcon"
    >
      <div
        class="inner filled"
      >
        <div
          class="addons leftIcon"
        >
          🚗
        </div>
        <div
          class="inputWrapper"
        >
          <label
            class="label"
          >
            <span
              class="labelInner"
            >
              Label
            </span>
          </label>
          <div
            class="input"
          >
            <input
              aria-label="Label"
              class="input hasLeftIcon"
              type="text"
              value="value"
            />
          </div>
        </div>
        <fieldset
          aria-hidden="true"
          class="fieldset"
        >
          <legend
            class="legend"
          >
            <span>
              Label
            </span>
          </legend>
        </fieldset>
        <div
          class="addons rightIcon"
        >
          🚗
        </div>
      </div>
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
